evap/staff/importers/json.py

```diff
@@ -0,0 +1,445 @@
+            "vote_start_datetime": evaluation_start_datetime,
+            "vote_end_date": evaluation_end_date,
+            "wait_for_grade_upload_before_publishing": wait_for_grade_upload_before_publishing,
+            "weight": weight,
+        }
+        evaluation, created = Evaluation.objects.get_or_create(
+            course=course,
+            cms_id=data["gguid"],
+            defaults=defaults,
+        )
+        if evaluation.state < Evaluation.State.APPROVED:
+            direct_changes = update_with_changes(evaluation, defaults)
+
+            participant_changes = set(evaluation.participants.all()) != set(participants)
+            evaluation.participants.set(participants)
+
+            any_lecturers_changed = False
+            if "lecturers" not in data:
+                self.statistics.warnings.append(
+                    WarningMessage(obj=evaluation.full_name, message="No contributors defined")
+                )
+            else:
+                for lecturer in data["lecturers"]:
+                    __, lecturer_created, lecturer_changes = self._import_contribution(evaluation, lecturer)
+                    if lecturer_changes or lecturer_created:
+                        any_lecturers_changed = True
+
+            if not created and (direct_changes or participant_changes or any_lecturers_changed):
+                self.statistics.updated_evaluations.append(evaluation)
+        else:
+            self.statistics.attempted_changes.append(evaluation)
+
+        if created:
+            self.statistics.new_evaluations.append(evaluation)
+
+        return evaluation
+
+    def _import_contribution(
+        self, evaluation: Evaluation, data: ImportRelated
+    ) -> tuple[Contribution | None, bool, dict[str, tuple[Any, Any]]]:
+        if data["gguid"] not in self.students_by_gguid:
+            return None, False, {}
+
+        user_profile = self.students_by_gguid[data["gguid"]]
+
+        if user_profile.email in settings.NON_RESPONSIBLE_USERS:
+            return None, False, {}
+
+        contribution, created, changes = update_or_create_with_changes(
+            Contribution,
+            evaluation=evaluation,
+            contributor=user_profile,
+        )
+        return contribution, created, changes
+
+    def _import_events(self, data: list[ImportEvent]) -> None:
+        # Divide in two lists so corresponding courses are imported before their exams
+        non_exam_events = (event for event in data if not event["isexam"])
+        exam_events = (event for event in data if event["isexam"])
+
+        for event in non_exam_events:
+            course = self._import_course(event)
+
+            self._import_evaluation(course, event)
+
+        unused_events = []
+        for event in exam_events:
+            if "relatedevents" not in event:
+                unused_events.append(event)
+                continue
+
+            exam_course = self.courses_by_gguid[event["relatedevents"][0]["gguid"]]
```

Comment:
Also here: What is the meaning of `exam_course`? Is it a special kind of course only for the exam, no right?

And this course could have come from an import from the `non_exam_events`, or from an existing course in the database right? So we amend the previous import with more data from the exam event here.

Can we add some more commentary to this section of the code? Without having the spec from the issue in mind, I find it hard to follow